scanpy/tests/test_clustering.py

```diff
@@ -20,6 +21,52 @@ def test_leiden_basic(adata_neighbors):
     sc.tl.leiden(adata_neighbors)
 
 
+def test_clustering_subset(adata_neighbors):
+    methods = (
+        (sc.tl.louvain, 'louvain'),
+        (sc.tl.leiden, 'leiden'),
+    )
+    for clustering, key in methods:
+        print('Clustering method ' + str(key))
+        clustering(adata_neighbors, key_added=key)
+        print(adata_neighbors)
+        print(adata_neighbors.obs[key].value_counts())
+
+        for c in np.unique(adata_neighbors.obs[key]):
+            print('Analyzing cluster ' + str(c))
+            cells_in_c = (adata_neighbors.obs[key] == c)
+            ncells_in_c = adata_neighbors.obs[key].value_counts().loc[c]
+            key_sub = str(key) + '_sub'
+            clustering(adata_neighbors, restrict_to=(key, [c]),
+                key_added=key_sub) 
+
+            # Old category should not be present
+            assert (c not in
+                adata_neighbors.obs.loc[:, key_sub].value_counts().index)
```

Comment:
Not a fan of the weird linebreaks nor all the redundant braces here and in other statements. Please do it like

```py
cells_in_c = adata_neighbors.obs[key] == c
```

and either

```py
clustering(adata_neighbors, restrict_to=(key, [c]), key_added=key_sub) 
```
or

```py
clustering(
    adata_neighbors,
    restrict_to=(key, [c]),
    key_added=key_sub,
)
```

Reply:
```python
adata_neighbors.obs.loc[:, key_sub].value_counts().index

Reply:
I partially rewrote this part too, using suggestions from both comments.